src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,41 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            System.out.println("Cache hit for userId: " + userId);
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Thread.sleep(1000);
+                String details = "Details for " + userId;
+                return new UserData(userId, details);
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Request processing interrupted", e);
+            }
+        }, executor).thenApply(userData -> {
+            cache.put(userId, userData);
+            System.out.println("Processed: " + userData);
+            return userData;
+        }).exceptionally(ex -> {
+            System.err.println("Failed to process request for userId: "
+                    + userId
+                    + ", Error: "
+                    + ex.getMessage());
+            return new UserData(userId, "Error occurred");
+        });
```

Comment:
The `exceptionally` block is handling exceptions by logging the error and returning a `UserData` object with an error message. Ensure that this behavior aligns with the application's requirements for error handling. If more sophisticated error handling is needed, consider implementing it here.